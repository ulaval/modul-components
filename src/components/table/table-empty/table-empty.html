<m-table-row>
    <slot v-if="hasSlot"></slot>
    <m-table-cell class="m-table-empty" v-else :colspan="columns">
        <m-icon class="m-table-empty-cell--icon" name="m-svg__no-data" size="55"></m-icon>
        <p>
            {{ i18nEmptyData }}
        </p>
    </m-table-cell>
</m-table-row>
