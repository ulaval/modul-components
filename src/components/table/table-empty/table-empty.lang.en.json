{
    "m-table-empty": {
        "empty": "No data to display"
    }
}
