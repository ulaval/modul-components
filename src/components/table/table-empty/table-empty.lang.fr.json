{
    "m-table-empty": {
        "empty": "Aucune donnée à afficher"
    }
}
