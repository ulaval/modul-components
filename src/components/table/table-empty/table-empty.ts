import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';

import { ModulVue } from '../../../utils/vue/vue';
import { TABLE_EMPTY_NAME } from '../../component-names';
import WithRender from './table-empty.html';

@WithRender
@Component
export class MTableEmpty extends ModulVue {

    @Prop()
    columns: number;

    public i18nEmptyData: string = this.$i18n.translate('m-table-empty:empty');

    public get hasSlot(): boolean {
        return !!this.$slots.default;
    }

}

const TableEmptyPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(TABLE_EMPTY_NAME, 'plugin.install');
        v.component(TABLE_EMPTY_NAME, MTableEmpty);
    }
};

export default TableEmptyPlugin;
