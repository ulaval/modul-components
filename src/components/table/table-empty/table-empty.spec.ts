import { shallow, Wrapper } from '@vue/test-utils';

import { renderComponent } from '../../../../tests/helpers/render';
import { MTableEmpty } from './table-empty';

let slots: any = {};
const slot: string = '<div>slot</div>';

let wrapper: Wrapper<MTableEmpty>;

const initializeShallowWrapper: any = () => {
    wrapper = shallow(MTableEmpty, {
        slots
    });
};

describe(`MTableEmpty`, () => {

    describe(`When no slot is provided`, () => {

        beforeEach(() => {
            slots = {};
            initializeShallowWrapper();
        });

        it(`Should render correctly`, () => {
            expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
        });

        it(`Then show the default empty message`, () => {
            expect(wrapper.vm.hasSlot).toBeFalsy();
        });

    });

    describe(`When a slot is provided`, () => {

        beforeEach(() => {
            slots = {
                default: slot
            };
            initializeShallowWrapper();
        });

        it(`Should render correctly`, () => {
            expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
        });

        it(`Then show the slot`, () => {
            expect(wrapper.vm.hasSlot).toBeTruthy();
        });

    });

});
