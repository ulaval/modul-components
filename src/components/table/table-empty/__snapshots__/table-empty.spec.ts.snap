// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MTableEmpty When a slot is provided Should render correctly 1`] = `
<m-table-row>
    <div>slot</div>
</m-table-row>
`;

exports[`MTableEmpty When no slot is provided Should render correctly 1`] = `
<m-table-row>
    <m-table-cell class="m-table-empty">
        <m-icon name="m-svg__heraldry" size="45" class="m-table-empty-cell--icon"></m-icon>
        <p>
            m-table-empty:empty
        </p>
    </m-table-cell>
</m-table-row>
`;
